libsolidity/codegen/MultiUseYulFunctionCollector.cpp

```diff
@@ -41,12 +41,14 @@ string MultiUseYulFunctionCollector::requestedFunctions()
 
 string MultiUseYulFunctionCollector::createFunction(string const& _name, function<string ()> const& _creator)
 {
-	if (!m_requestedFunctions.count(_name))
+	if (!m_requestedFunctions.count(_name) && !m_beingGeneratedFunctions.count(_name))
 	{
+		m_beingGeneratedFunctions.emplace(_name);
```

Comment:
Maybe easier to do
```
// Guard needed for generating recursive functions.
m_requestedFunctions[_name] = "STUB";
```
And get rid of ``m_beingGeneratedFunctions``?

But hm... I'm actually not sure whether that would be better or worse :-)...

Reply:
So I'm also fine with keeping ``m_beingGeneratedFunctions`` and documenting it.
But then you should probably also at least add ``solAssert(m_beingGeneratedFunctions.empty(), "");`` to ``requestedFunctions()``?

If we go the other way and add "STUB" entries, we could have a ``solAssert(f.second != "STUB", "");`` there - or we could use the empty string instead of "STUB" and use ``solAssert(!f.second.empty(), "");`` there...

Reply:
I went with "STUB" entries